static/js/stream_ui_updates.js

```diff
@@ -143,7 +143,11 @@ exports.update_stream_privacy_type_icon = function (sub) {
 
 exports.update_stream_subscription_type_text = function (sub) {
     const stream_settings = stream_edit.settings_for_sub(sub);
-    const html = render_subscription_type(sub);
+    const template_data = sub;
```

Comment:
You should do make a new object based on a copy of sub to do this sort of template work.  The `sub` objects returned by `settings_for_sub` are singletons, so mutating them is a bad idea.